### ISSUE TITLE
Scaling down a crewed part makes the crew go M.I.A. and the crew capacity is not restored later!

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/TweakScale#247]
Fellow Kerbonaut [robi_243 ](https://forum.kerbalspaceprogram.com/index.php?/profile/221308-robi_243/) [found something](https://forum.kerbalspaceprogram.com/index.php?/topic/208059-tweakscale-problem/#comment-4128043) on TweakScale and (probably) newer KSP releases.

If you downscale a Mark 1 Cockpit two notches, two things happens:

1. If the part is crewed, the Kerbals go M.I.A.!!!! 
2. Once you scale back the part, the crew capacity is not restored!!

~~I'm absolutely sure I had tested the Item 2 when I first released this feature - but don't remember if I had tested the item 1.~~

Me and my memory. :P On KSP 1.7.3, the problem 1 **does not happens**, but the problem 2 does. So, and assuming I really tested the item 2 at that times (I still think I did), the item 2 is a new bug that is my fault - besides the bug 1 **being not**. :D 

In a way or another, I need to handle this somehow.

**Comment by Lisias** [comment]
The bug 1 **is not my fault**. On KSP 1.7.3, the Kerbal is recovered to the roster as expected.

But the bug 2 appears to be my fault, because it happened on KSP 1.7.3 - and I have the memory of testing this on it in the past.

**Comment by Lisias** [comment]
KSP 1.8.1 behave **exactly** as 1.7.3.

**Comment by Lisias** [comment]
KSP 1.9.1 behave **exactly** as the previous ones.

**Comment by Lisias** [comment]
KSP 1.11.2 also behave **exactly** as the above mentioned ones. I jumped over 1.10.1 and it appears obvious to me that it would behave the same.

This is a new crap on KSP 1.12.x. "Thanks", Squad. :/

**Comment by Lisias** [comment]
Saving the craft and reloading is also a work around for bug 2.

**Comment by Lisias** [comment]
Oukey, it was **totally** my fault after all.

On commit https://github.com/net-lisias-ksp/TweakScale/commit/77ee66746c2599cc2bbebb0bb220b526f7f0292a I ended up short circuiting something that I should not, in special aborting the execution of the Crew Scaling if the current Crew Scale was equal or bigger than the prefab (as [KSP is screwing with TweakScale on Crew since 1.8.0](https://forum.kerbalspaceprogram.com/index.php?/topic/179030-ksp-130-tweakscale-under-lisias-management-24612-2022-0429/page/48/#comment-3702846)). But, **obviously** I should not had used the ">=" operator, as when you are scaling back a part from a smaller scale, you will need to rebuild the kraken damned crew roster of the part even if the new size is the same as the original size, damn it!

The reason the Crew is going to M.I.A. on KSp 1.12.x is also due that (and the reason I removed the "Not My Fault" label, as it ended up being my fault after all). That short circuit prevented the crew resizing code to run on some borderline situations (as this one) and so the crew was not being correctly removed form the part. On previous KSP versions this mishap was being tolerated, but on KSP 1.12.x some code change removed that tolerance and the Kerbal was declared M.I.A .

(sigh)

Anyway, **properly** fixed this time on commit https://github.com/net-lisias-ksp/TweakScale/commit/53686505685f6002a7b7bfe0d45750351937db70
